packages/puppeteer-core/src/common/Binding.ts

```diff
@@ -0,0 +1,107 @@
+import {isErrorLike} from '../util/ErrorLike.js';
+import {ExecutionContext} from './ExecutionContext.js';
+
+/**
+ * @internal
+ */
+export class Binding {
+  #name: string;
+  #fn: (...args: unknown[]) => unknown;
+  constructor(name: string, fn: (...args: unknown[]) => unknown) {
+    this.#name = name;
+    this.#fn = fn;
+  }
+
+  /**
+   * @param throwError - Whether to throw the error instead of sending it to the
+   * binding.
+   */
+  async call(
+    context: ExecutionContext,
+    seq: number,
+    args: unknown[]
+  ): Promise<void> {
+    try {
+      // Getting non-trivial arguments.
+      const handles = await context.evaluateHandle(
```

Comment:
so we do those extra calls on every binding call even if there were no non-primitive arguments?

Reply:
Done.